tests/admin_views/tests.py

```diff
@@ -1472,21 +1477,67 @@ def test_conditionally_show_add_section_link(self):
         login_url = reverse('admin:login') + '?next=/test_admin/admin/'
         # Set up and log in user.
         url = '/test_admin/admin/admin_views/article/add/'
-        add_link_text = ' class="add-another"'
+        add_link_text = 'add_id_section'
         self.client.get('/test_admin/admin/')
         self.client.post(login_url, self.adduser_login)
-        # The add user can't add sections yet, so they shouldn't see the "add
+        # The user can't add sections yet, so they shouldn't see the "add
         # section" link.
         response = self.client.get(url)
         self.assertNotContains(response, add_link_text)
-        # Allow the add user to add sections too. Now they can see the "add
+        # Allow the user to add sections too. Now they can see the "add
         # section" link.
-        add_user = User.objects.get(username='adduser')
+        user = User.objects.get(username='adduser')
         perm = get_perm(Section, get_permission_codename('add', Section._meta))
-        add_user.user_permissions.add(perm)
+        user.user_permissions.add(perm)
         response = self.client.get(url)
         self.assertContains(response, add_link_text)
 
+    def test_conditionally_show_change_section_link(self):
+        """
+        The foreign key widget should only show the "change related" button if
+        the user has permission to change that related item.
+        """
+        login_url = reverse('admin:login') + '?next=/test_admin/admin/'
+        # Set up and log in user.
+        url = '/test_admin/admin/admin_views/article/add/'
+        change_link_text = 'change_id_section'
+        self.client.get('/test_admin/admin/')
```

Comment:
does the get() have a purpose? It seems you've copied this from the other tests, but I think we can clean this up cause I don't see a purpose for that. Similarly for `'?next=/test_admin/admin/'` in the login_url. I wonder if we should use `self.client.login()` instead as well. Feel free to create a separate ticket for these issues though to avoid bogging this down.